Adding Validations to asazure.json

athipp opened the pull request:
Adding validations to resourcegroupname and analysisservername

athipp committed:
Adding Validations to asazure.json

amarzavery commented on arm-asazure/2016-05-06/swagger/asazure.json:
```diff
@@ -418,6 +430,7 @@
       "in": "path",
       "required": true,
       "type": "string",
+      "pattern": "^[-\\w\\._]+$",
```
`"pattern": "^[-\\w\\._\\(\\)]+$",`

tbombach replied:
@amarzavery We should standardize this and make it part of good patterns/the skeleton Swagger file I'm working on.

amarzavery replied:
It would be nice to define commonly used model definitions and parameters in a common file and ask everyone to reference from there. That makes things consistent.

athipp committed:
Fixed some bugs

- regex for resource group
- "resourceType" is acutally just "type" and can be moved t base type
"resource"
- "serverAdministrators" is "asAdministrators"

AutorestCI commented:
No modification for NodeJS

AutorestCI commented:
No modification for Ruby

AutorestCI commented:
No modification for Python
